### PR TITLE
React interactions collection list - service filter

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description of change

This add the services filter to the new React interactions collection list.

## Test instructions

1. Go to `/interactions/react`
2. Toggle the services checkboxes
3. View results

## Screenshots
Screenshot 2021-06-27 at 11 05 23

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/interactions/client/filters.js [hunk 1]
```diff
@@ -2,28 +2,38 @@ import { KIND_OPTIONS, LABELS } from './constants'
...
-    categoryLabel: LABELS.dateAfter,
-  }),
-  selectedDatesBefore: buildDatesFilter({
-    value: queryParams.date_before,
-    categoryLabel: LABELS.dateBefore,
-  }),
-})
+export const buildSelectedFilters = (
+  { queryParams, selectedAdvisers },
+  { serviceOptions }
```
Comment (paulgain) [inline src/apps/interactions/client/filters.js:7]: This is verbose and unnecessary, we should't be destructuring here, we should just pass the variables in as is. It makes no sense to create an object containing two values only to destruct them immediately on function invocation.

**Review by paulgain — COMMENTED** [review]
LGTM, just a few minor comments.

File: src/apps/interactions/client/state.js [hunk 2]
```diff
@@ -23,11 +24,13 @@ export const state2props = ({ router, ...state }) => {
...
-  const selectedFilters = buildSelectedFilters({
-    queryParams,
-    metadata,
-    selectedAdvisers,
-  })
+  const selectedFilters = buildSelectedFilters(
+    {
+      queryParams,
+      selectedAdvisers,
+    },
```
[the remark this answers is not in the document]
Comment (tomkingston) [reply src/apps/interactions/client/state.js:31]: So, in general I would say the gain would be that you know what arguments you are passing into the other function without having to refer back to that function. However, in this, case metadata should be included within the first object like this:
```
const selectedFilters = buildSelectedFilters(
    {
      queryParams,
      metadata,
      selectedAdvisers,
    }
)
```

**Review by tomkingston — APPROVED** [review]
Approved subject to Paul's comments

File: src/apps/interactions/client/state.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulgain) [reply src/apps/interactions/client/state.js:31]: In this instance there is no gain, I'd much to prefer to read the second implementation.

```javascript
const selectedFilters = buildSelectedFilters(
    {
      queryParams,
      metadata,
      selectedAdvisers,
    }
)

buildSelectedFilters({ queryParams, metadata, selectedAdvisers})
```

```javascript
const selectedFilters = buildSelectedFilters(
    queryParams,
    metadata,
    selectedAdvisers
)

buildSelectedFilters(queryParams, metadata, selectedAdvisers)
```

File: test/functional/cypress/specs/interaction/filter-react-spec.js [hunk 3]
```diff
@@ -45,6 +48,7 @@ const adviser = {
   id: adviserId,
   name: 'Barry Oling',
 }
+const services = servicesListFaker(10)
```
[the remark this answers is not in the document]
Comment (ian-leggett) [reply test/functional/cypress/specs/interaction/filter-react-spec.js]: Good point not sure why I created 10

**Review by paulgain — APPROVED** [review]
Looking good 👍

**Pull request merged**